packages/core/src/lib/createSystem.ts

```diff
@@ -108,6 +108,8 @@ export function createSystem(config: KeystoneConfig) {
     adminMeta,
     getKeystone: (prismaModule: PrismaModule) => {
       const prePrismaClient = new prismaModule.PrismaClient({
+        adapter:
+          typeof config.db.prismaAdapter === 'function' ? config.db.prismaAdapter() : undefined,
```

Comment:
Maybe we should support a custom `new prismaModule.PrismaClient` instead

Reply:
If users did that, we could either pass the `datasources` and `log` parameters to them, or alternatively prevent them through a sum type.

Reply:
@dcousens yes that makes totally sense 👍 will do that.

That also makes it possible to update the prisma client so users can use their own prisma client extensions (already tried to do that on a project but only could make it work with the not recommended Prisma Middleware approach)

Updating the PR asap 👍

Reply:
@dcousens could you check if this is going the desired direction?

https://github.com/keystonejs/keystone/pull/8847/commits/2d12d6bd182def439c9d8bde9d6c79ea8a4fbb98

Reply:
LGTM @mmachatschek, I'll probably refine and bike shed, but if you wanted to ensure this is working for your usecases, I'm happy to help push it across the line for the next major.

Reply:
@dcousens would be great if you could push that across the line. I can assist in any way. I haven't dived into the testing stuff of this repo. Happy to help in any direction